fetchmail_attach_from_folder/models/fetchmail_server_folder.py

```diff
@@ -227,11 +233,30 @@ def apply_matching(self, connection, msgid):
                     thread_id = match.id
                     self.attach_mail(match, message_dict)
         matched = True if thread_id else False
+        if matched:
+            self.run_server_action(thread_id)
         self.update_msg(connection, msgid, matched=matched)
         if self.archive_path:
             self._archive_msg(connection, msgid)
         return thread_id  # Can be None if no match found.
 
+    def run_server_action(self, matched_object_ids):
```

Comment:
Please add a test for this method, to increase instead of decrease the test coverage.

Reply:
tests added